src/main/java/org/zalando/baigan/service/aws/S3FileLoader.java

```diff
@@ -0,0 +1,100 @@
+package org.zalando.baigan.service.aws;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.AWSKMSClient;
+import com.amazonaws.services.kms.model.DecryptRequest;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.google.common.io.BaseEncoding;
+
+import javax.annotation.Nonnull;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+/* Provides transparent content decryption of encrypted configuration content using AWS KMS. All configuration values
+ * surrounded by {@value #KMS_START_TAG} and {@value #KMS_END_TAG} are decrypted automatically. The content must
+ * be Base64 encoded and the decrypted content is interpreted as an UTF-8 encoded string. */
+
+public class S3FileLoader {
+
+    private static final String KMS_START_TAG = "awskms{";
```

Comment:
I many other, related/similar, tools there's a standard of just prefixing with `awk:kms:`
For example, an encrypted key value would contain:
  `aws:kms:fooBar` if the encrypted value was fooBar.
Also seems a bit less complex than having an opening and closing tag.